core/src/services/gcs/backend.rs

```diff
@@ -74,6 +74,8 @@ pub struct GcsConfig {
     pub disable_vm_metadata: bool,
     /// Disable loading configuration from the environment.
     pub disable_config_load: bool,
+    /// A Google Cloud OAuth2 token.
+    pub token: String,
```

Comment:
I see that `token` is used in a few other places and I've opted for a simple `String` here.

I might be oversimplifying this though too, feel free to let me know :+1:

Reply:
Using a string looks good to me! We can use `Option<String>` to allow the builder to distinguish between its `None` and default value. At the same time, we can provide an example of how this token appears.

We should clarify that this value will take precedence over the `credential`. If a token has been set, we will not attempt to fetch a token via the credential anymore.